### ISSUE TITLE
关于制作腹部CT的pkl数据集问题

### ISSUE THREAD

**Issue by acav485** [x-xc/SACB_Net#5]
请问作者是否方便给出制作腹部CT的pkl数据集的代码，小白实在是写不出来这部分的代码，感谢！！！

**Comment by x-xc** [comment]
import torchvision.transforms as T
import pickle
import torchio as tio
import glob
import os

data_pth = sorted(glob.glob('./imagesTr/*'))
label_pth =  sorted(glob.glob('./labelsTr/*'))
def pksave(img, label, save_path):
    with open(save_path, 'wb') as f:
        pickle.dump((img, label), f)
save_path = '/bask/homes/x/xxc142/xxc/dataset_all/AbdomenCTCT/processeced/' 
os.makedirs(save_path, exist_ok=True)  

for i in range(len(data_pth)):
        print(data_pth[i].split('/')[-1], label_pth[i].split('/')[-1])
        data = tio.ScalarImage(data_pth[i])
        label = tio.ScalarImage(label_pth[i])
        min, max = -1000, 1000 # , in_min_max=(min, max)

        trans = T.Compose([
                tio.Clamp(out_min=min, out_max=max),
                tio.RescaleIntensity(out_min_max=(0, 1)),
                tio.CropOrPad((192, 160, 224))])
        trans2 = T.Compose([tio.CropOrPad((192, 160, 224))])
        ct_normalized = trans(data).data.numpy().astype('float32')[0]
        label_ = trans2(label).data.numpy().astype('uint16')[0]
        pksave(ct_normalized,label_, save_path=save_path+'subject_%02d.pkl'%(i+1))

**Comment by acav485** [comment]
非常感谢您！！！
